karapace/instrumentation/prometheus.py

```diff
@@ -0,0 +1,100 @@
+"""
+karapace - prometheus instrumentation
+
+Copyright (c) 2024 Aiven Ltd
+See LICENSE for details
+"""
+# mypy: disable-error-code="call-overload"
+
+from __future__ import annotations
+
+from aiohttp.web import middleware, Request, Response
+from karapace.rapu import RestApp
+from prometheus_client import CollectorRegistry, Counter, Gauge, generate_latest, Histogram
+from typing import Awaitable, Callable, ClassVar
+
+import logging
+import time
+
+LOG = logging.getLogger(__name__)
+
+
+class PrometheusInstrumentation:
+    METRICS_ENDPOINT_PATH: ClassVar[str] = "/metrics"
+    START_TIME_REQUEST_KEY: ClassVar[str] = "start_time"
+
+    registry: ClassVar[CollectorRegistry] = CollectorRegistry()
+
+    karapace_http_requests_total: ClassVar[Counter] = Counter(
+        registry=registry,
+        name="karapace_http_requests_total",
+        documentation="Total Request Count for HTTP/TCP Protocol",
+        labelnames=("method", "path", "status"),
+    )
+
+    karapace_http_requests_latency_seconds: ClassVar[Histogram] = Histogram(
+        registry=registry,
+        name="karapace_http_requests_latency_seconds",
+        documentation="Request Duration for HTTP/TCP Protocol",
+        labelnames=("method", "path"),
+    )
+
+    karapace_http_requests_in_progress: ClassVar[Gauge] = Gauge(
+        registry=registry,
+        name="karapace_http_requests_in_progress",
+        documentation="Request Duration for HTTP/TCP Protocol",
```

Comment:
**[issue]**
small copy paste error here on the `documentation`, i'd fix after review.